### DISCUSSION TITLE
train model ABINet using visual part

### DISCUSSION THREAD

**Discussion by Aminfaraji** [PaddlePaddle/PaddleOCR#14474] [Q&A]
Hello, I hope you're doing well. I want to use the ABINet model, but I don't want to use the language model part. In the config file, I set the 'use_lang' parameter to false, but I get an error. I only want to use the visual model part. I would appreciate your guidance.
@jzhang533 @kerneltravel @wencan @sdcb @ZeyuChen @Topdu

**Comment by GreatV** [comment]
Hello! To train the ABINet model using only the visual part and disable the language model, setting the `use_lang` parameter to `false` in the configuration file is the correct approach. However, if you are encountering an error, it might be due to one of the following reasons:

---

### 1. **Potential Configuration Issues**
   - Ensure you have correctly updated the configuration file (`rec_r45_abinet.yml`) to disable the language model. Specifically, locate the section where `use_lang` is defined and set it to `false`:
     ```yaml
     use_lang: false
     ```
   - Double-check that all other parameters related to the language model are either commented out or not active to avoid conflicts.

---

### 2. **Code Dependencies**
   - The error may occur if the codebase is trying to access components of the language model despite it being disabled. Review the training script (`tools/train.py`) to ensure no hardcoded language model dependencies exist.
   - If necessary, update the relevant sections of the code to skip language model initialization and operations when `use_lang` is `false`.

---

### 3. **Missing Pre-trained Model or Dependencies**
   - If you're using a pre-trained model, ensure it was trained without the language model or is compatible with the `use_lang: false` configuration.
   - Use the pre-trained model provided in the [ABINet repository](https://paddleocr.bj.bcebos.com/rec_r45_abinet_train.tar) or train a new model specifically for the visual part.

---

### 4. **Known Issues and Debugging**
   - If the issue persists, you can debug by adding print statements or logging around areas where the error occurs. This will help identify if the visual model is attempting to interact with the disabled language model.
   - Check the PaddleOCR GitHub repository for related issues or discussions. For example, you might find help in [PaddleOCR Issues](https://github.com/PaddlePaddle/PaddleOCR/issues).

---

### 5. **Alternative Approach**
   - If modifying the configuration and code does not resolve the issue, you can manually comment out or remove all language model-related components in the training code. This includes:
     - Language model initialization.
     - Loss functions or metrics related to the language model.
     - Any forward passes through the language model.

---

### Example Training Command:
After updating the configuration file, train the model using:
```bash
python3 tools/train.py -c configs/rec/rec_r45_abinet.yml
```

If using multi-GPU, use:
```bash
python3 -m paddle.distributed.launch --gpus '0,1,2,3' tools/train.py -c configs/rec/rec_r45_abinet.yml
```

---

### 6. **Error Context**
If you can share the specific error message you're encountering, I can provide more targeted guidance!

Let me know if you need further assistance!

---
*Response generated by [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*

**Reply by Aminfaraji** [reply]
`ValueError: (InvalidArgument) Input(Logits) and Input(Label) should in same shape in dimensions except axis.
  [Hint: Expected logits_dims[i] == labels_dims[i], but received logits_dims[i]:832 != labels_dims[i]:288.] (at ../paddle/phi/infermeta/binary.cc:936)
`
